goaxios.go

```diff
@@ -11,6 +11,9 @@ import (
 // a wrapper around Go's *http.Request ojbect to make it faster to run REST http requests.
 // It returns the *http.Response object, the response body as byte, the unmarshalled response body and an error object (if any or nil)
 func (ga *GoAxios) RunRest() (*http.Response, []byte, interface{}, error) {
+	if ga.Interceptor.Request != nil {
+		ga = ga.Interceptor.Request(ga)
```

Comment:
If `ga` is a pointer...do we really need to do a re-assignment to `ga` again after calling the interceptor?